### PR TITLE
refactor(Popup): create first internal component popup used by tooltip and popover

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Refactor

#### What is expected?

Creation of a new folder `src/internalComponents`. This folder will contains all components that are being used within the package and will never be exported outside of it. The goal is to have a centralised place for reusable components.

In this PR we created `Popup` a component that show on screen as popup depending on the location of the children. This component is used by `Tooltip` but also by `Popover` and will later be used for `SelectInput`, `Menu` and any other components needing `Popup` behavior.

### CODE REVIEW

File: packages/ui/src/components/Tooltip/index.tsx [hunk 1]
```diff
@@ -1,100 +1,6 @@
...
-`
-
-const StyledChildrenContainer = styled.div`
-  display: inherit;
-`
+import type { ComponentProps, ReactNode, Ref, RefObject } from 'react'
+import { forwardRef } from 'react'
+import { Popup } from '../../internalComponents'
 
 type TooltipProps = {
```
Comment (chambo-e) [inline packages/ui/src/components/Tooltip/index.tsx:5]: Can we reuse Popup type here ?

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: packages/ui/src/internalComponents/Popup/index.tsx [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+  tooltipPosition: string
+}
+
+type StyledTooltipProps = {
+  maxWidth: number
+  positions: PositionsType
+  reverseAnimation: boolean
+}
+
+const StyledTooltip = styled.div<StyledTooltipProps>`
```
Comment (philibea) [inline packages/ui/src/internalComponents/Popup/index.tsx:59]: shouldForwardProp ?

File: packages/ui/src/internalComponents/Popup/__tests__/__tests__/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+import type { ComponentProps } from 'react'
+import {
+  renderWithTheme,
+  shouldMatchEmotionSnapshot,
+} from '../../../../../.jest/helpers'
+import { Popup } from '../../index'
+
+describe('Popup', () => {
+  test('should render correctly', () =>
+    shouldMatchEmotionSnapshot(<Popup text="test">Hover me</Popup>))
```
Comment (philibea) [inline packages/ui/src/internalComponents/Popup/__tests__/__tests__/index.test.tsx:12]: shouldMatchEmotionSnapshotWithPortal

File: packages/ui/src/internalComponents/Popup/__tests__/__tests__/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+  test('should render correctly without text', () =>
+    shouldMatchEmotionSnapshot(<Popup>Hover me</Popup>))
```
Comment (philibea) [inline packages/ui/src/internalComponents/Popup/__tests__/__tests__/index.test.tsx:15]: shouldMatchEmotionSnapshotWithPortal

File: packages/ui/src/components/Tooltip/__tests__/__snapshots__/index.test.tsx.snap [hunk 5]
```diff
@@ -8,7 +8,7 @@ exports[`Tooltip should render correctly 1`] = `
 
 <div
```
Comment (philibea) [inline packages/ui/src/components/Tooltip/__tests__/__snapshots__/index.test.tsx.snap:9]: check here if should use shouldMatchEmotionSnapshotWithPortal instead :)

**Review by chambo-e — APPROVED** [review]

**Review by JulienSaguez — APPROVED** [review]

**Review by philibea — APPROVED** [review]

**Pull request merged**